opencog/cython/opencog/bindlink.pyx

```diff
@@ -1,15 +1,21 @@
-from opencog.atomspace cimport Atom, AtomSpace, TruthValue
-from opencog.atomspace cimport cHandle, cAtomSpace, cTruthValue
+from opencog.atomspace cimport Value, Atom, AtomSpace, TruthValue
+from opencog.atomspace cimport cAtomSpace, cTruthValue
 from opencog.atomspace cimport tv_ptr, strength_t, count_t
 from opencog.atomspace cimport handle_cast
 from cython.operator cimport dereference as deref
 
+from opencog.atomspace import is_a, types
+
 def execute_atom(AtomSpace atomspace, Atom atom):
     if atom == None: raise ValueError("execute_atom atom is: None")
     cdef cValuePtr c_value_ptr = c_execute_atom(atomspace.atomspace,
                                            deref(atom.handle))
-    cdef cHandle c_result = handle_cast(c_value_ptr)
-    return Atom.createAtom(c_result, atomspace)
+
+    if is_a(deref(c_value_ptr).get_type(), types.Atom):
```

Comment:
Such implementation will not return correct class of Atom or Value. Please use `create_value_by_type` instead, see:
https://github.com/opencog/atomspace/blob/5b47b937b7cb1d03785042bf636ae0e2e90a5444/opencog/cython/opencog/atom.pyx#L71-L78

It handles both Atom and Value cases.

Reply:
I updated the code to use create_value_by_type and updated the test to check that value.to_list()  works as expected.